source/dofs/dof_handler.cc

```diff
@@ -1104,9 +1104,16 @@ DoFHandler<dim,spacedim>::renumber_dofs (const std::vector<types::global_dof_ind
 {
   Assert(levels.size()>0, ExcMessage("You need to distribute DoFs before you can renumber them."));
 
-  AssertDimension (new_numbers.size(), n_locally_owned_dofs());
-
 #ifdef DEBUG
+  if (dynamic_cast<const parallel::shared::Triangulation< dim, spacedim>*> (&*tria) != nullptr)
+    {
+      AssertDimension (new_numbers.size(), n_dofs());
+    }
+  else if (dynamic_cast<const parallel::distributed::Triangulation< dim, spacedim >*> (&*tria) != nullptr)
+    {
+      AssertDimension (new_numbers.size(), n_locally_owned_dofs());
+    }
```

Comment:
add another `else` (no `if`) here where you test the sequential case where the size needs to be `n_dofs()`.